bundle/terranova/tnresources/all_test.go

```diff
@@ -0,0 +1,159 @@
+package tnresources
+
+import (
+	"context"
+	"reflect"
+	"testing"
+
+	"github.com/databricks/cli/bundle/config/resources"
+	"github.com/databricks/cli/libs/structdiff/structpath"
+	"github.com/databricks/cli/libs/structwalk"
+	"github.com/databricks/cli/libs/testserver"
+	"github.com/databricks/databricks-sdk-go"
+	"github.com/databricks/databricks-sdk-go/service/apps"
+	"github.com/databricks/databricks-sdk-go/service/catalog"
+	"github.com/databricks/databricks-sdk-go/service/database"
+	"github.com/stretchr/testify/require"
+)
+
+var testConfig map[string]any = map[string]any{
+	"apps": &resources.App{
+		App: apps.App{
+			Name: "myapp",
+		},
+	},
+
+	"schemas": &resources.Schema{
+		CreateSchema: catalog.CreateSchema{
+			CatalogName: "main",
+			Name:        "myschema",
+		},
+	},
+
+	"volumes": &resources.Volume{
+		CreateVolumeRequestContent: catalog.CreateVolumeRequestContent{
+			CatalogName: "main",
+			SchemaName:  "myschema",
+			Name:        "myvolume",
+		},
+	},
+
+	"database_instances": &resources.DatabaseInstance{
+		DatabaseInstance: database.DatabaseInstance{
+			Name: "mydbinstance",
+		},
+	},
+
+	"database_catalogs": &resources.DatabaseCatalog{
+		DatabaseCatalog: database.DatabaseCatalog{
+			Name:                 "mydbcatalog",
+			DatabaseInstanceName: "mydbinstance1",
+		},
+	},
+
+	"synced_database_tables": &resources.SyncedDatabaseTable{
+		SyncedDatabaseTable: database.SyncedDatabaseTable{
+			Name: "main.myschema.my_synced_table",
+		},
+	},
+}
+
+type prepareWorkspace func(client *databricks.WorkspaceClient) error
+
+// some resource require other resources to exist
+var testDeps = map[string]prepareWorkspace{
+	"database_catalogs": func(client *databricks.WorkspaceClient) error {
+		_, err := client.Database.CreateDatabaseInstance(context.Background(), database.CreateDatabaseInstanceRequest{
+			DatabaseInstance: database.DatabaseInstance{
+				Name: "mydbinstance1",
+			},
+		})
+		return err
+	},
+}
+
+func TestAll(t *testing.T) {
+	_, client := setupTestServerClient(t)
+
+	for group, resource := range SupportedResources {
+		t.Run(group, func(t *testing.T) {
+			adapter, err := NewAdapter(resource, client)
+			require.NoError(t, err)
+			require.NotNil(t, adapter)
+
+			testCRUD(t, group, adapter, client)
+		})
+	}
+
+	m, err := InitAll(client)
+	require.NoError(t, err)
+	require.Len(t, m, len(SupportedResources))
+}
+
+func testCRUD(t *testing.T, group string, adapter *Adapter, client *databricks.WorkspaceClient) {
```

Comment:
Technically we only test create and update here.

Reply:
yeah, will expand this in follow up to test all methods.